atom/browser/api/atom_api_window.cc

```diff
@@ -644,6 +644,16 @@ double Window::GetOpacity() {
   return window_->GetOpacity();
 }
 
+void Window::SetShape(const std::vector<gfx::Rect>& rects) {
+#if !defined(OS_MACOSX)
+  auto native_window_views = static_cast<NativeWindowViews*>(window_.get());
+  auto shape = base::MakeUnique<SkRegion>();
+  for (const auto& rect : rects)
+    shape->op(gfx::RectToSkIRect(rect), SkRegion::kUnion_Op);
```

Comment:
The `SkRegion::setRects()` documentation suggests it be used instead of the above:

```cpp
    /**
     *  Set this region to the union of an array of rects. This is generally
     *  faster than calling region.op(rect, kUnion_Op) in a loop. If count is
     *  0, then this region is set to the empty region.
     *  @return true if the resulting region is non-empty
     */
    bool setRects(const SkIRect rects[], int count);
```

Reply:
Huh! I copied this code from the [DesktopWindowTreeHostX11::SetShape](https://chromium.googlesource.com/chromium/src/+/master/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc#698) code.